airbrakes/data_handling/logger.py

```diff
@@ -243,8 +274,14 @@ def _logging_loop(self) -> None:
             while True:
                 # Get a message from the queue (this will block until a message is available)
                 # Because there's no timeout, it will wait indefinitely until it gets a message.
-                message_fields = self._log_queue.get()
+                message_fields: list[LoggedDataPacket | Literal["STOP"]] = self._log_queue.get_many(
+                    timeout=MAX_GET_TIMEOUT_SECONDS
+                )
                 # If the message is the stop signal, break out of the loop
-                if message_fields == STOP_SIGNAL:
-                    break
-                writer.writerow(Logger._truncate_floats(message_fields))
+                for message_field in message_fields:
+                    if message_field == STOP_SIGNAL:
+                        break
+                    writer.writerow(Logger._truncate_floats(message_field))
+                else:
```

Comment:
No, it is for the `for` statement. In python, the `else` block of a `for` statement executes if and only if the loop is NOT terminated by a break statement. 

We need to do this because the inner loop's `break` statement will not break the outer loop also.

Reply:
though I guess I can just `return` instead and make it simpler